src/plugins/data_source/server/legacy/configure_legacy_client.ts

```diff
@@ -54,26 +64,47 @@ export const configureLegacyClient = async (
  * @returns child client.
  */
 const getQueryClient = async (
-  rootClient: Client,
-  dataSource: DataSourceAttributes,
+  dataSourceAttr: DataSourceAttributes,
   cryptography: CryptographyServiceSetup,
-  { endpoint, clientParams, options }: LegacyClientCallAPIParams
+  { endpoint, clientParams, options }: LegacyClientCallAPIParams,
+  { addClientToPool }: OpenSearchClientPoolSetup,
+  rootClient?: Client
 ) => {
-  const authType = dataSource.auth.type;
+  const {
+    auth: { type },
+    endpoint: nodeUrl,
+    lastUpdatedTime,
+  } = dataSourceAttr;
 
-  switch (authType) {
+  switch (type) {
     case AuthType.NoAuth:
-      return await (callAPI.bind(null, rootClient) as LegacyAPICaller)(
+      addClientToPool(nodeUrl, type, rootClient!);
+
+      return await (callAPI.bind(null, rootClient!) as LegacyAPICaller)(
         endpoint,
         clientParams,
         options
       );
+
     case AuthType.UsernamePasswordType:
-      const credential = await getCredential(dataSource, cryptography);
-      return getBasicAuthClient(rootClient, { endpoint, clientParams, options }, credential);
+      addClientToPool(nodeUrl, type, rootClient!);
+
+      const credential = await getCredential(dataSourceAttr, cryptography);
+      return getBasicAuthClient(rootClient!, { endpoint, clientParams, options }, credential);
+
+    case AuthType.SigV4:
+      const awsCredential = await getAWSCredential(dataSourceAttr, cryptography);
+      const awsClient = rootClient ? rootClient : getAWSClient(awsCredential, nodeUrl);
+      addClientToPool(nodeUrl + lastUpdatedTime, type, awsClient);
```

Comment:
Could we add some comments explaining why include lastUpdatedTime in cache key? This will be helpful as a reminder in case later we update the credentials without updating the saved object. 

Should we also create an issue in OpenSearch.js to support childClient?

Reply:
added comments both here and in client_pool.ts to explain